go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -667,6 +667,32 @@ func (se *Engine) GetSchema() map[string]*Table {
 	return tables
 }
 
+// MarshalMinimalSchema returns a protobuf encoded binlogdata.MinimalSchema
+func (se *Engine) MarshalMinimalSchema() ([]byte, error) {
+	se.mu.Lock()
+	defer se.mu.Unlock()
+	dbSchema := &binlogdatapb.MinimalSchema{
+		Tables: []*binlogdatapb.MinimalTable{},
+	}
```

Comment:
We know the capacity we need at this point so we should do a single allocation for the needed capacity:
```suggestion
	dbSchema := &binlogdatapb.MinimalSchema{
		Tables: make([]*binlogdatapb.MinimalTable, 0, len(se.tables)),
	}
```

Reply:
done in https://github.com/vitessio/vitess/pull/13045/commits/ee3c5f74f1e95335868b0294f21fd9af59c90689